### PR TITLE
Handle .docx, .xlsx, and .pptx with extra magic

### PR DESCRIPTION
**Pull request by scpike**
The freedesktop magic doesn't work for the Office 2007+ files, it
detects them all as application/zip. I have magic which detects them,
but it requires looking at 2,000 byte ranges, too much for their
project (which has to support things like file browsers). Since you
already warn that `by_magic` is slow, it'd be nice to support it here
at least.

This would allow the paperclip project to use mimemagic and stop
relying on the `file` binary:

  - https://github.com/thoughtbot/paperclip/issues/1530#issuecomment-43375497

Support is enabled through an overlay, required with

    require 'mimemagic/overlay'

Documentation is in the README

### CODE REVIEW

**Comment by scpike** [comment]
@minad new PR for your suggestions in #19.

File: test/mimemagic_test.rb [hunk 1]
```diff
@@ -50,9 +50,16 @@
...
+    file = "test/files/application.vnd.openxmlformats-officedocument.spreadsheetml.sheet"
+    MimeMagic.by_magic(File.read(file)).should.equal "application/zip"
+    MimeMagic.by_magic(File.open(file, 'rb')).should.equal "application/zip"
+  end
+
   it 'should recognize by magic' do
+    require "mimemagic/overlay"
     Dir['test/files/*'].each do |file|
-      mime = file[11..-1].gsub('.', '/')
+      mime = file[11..-1].sub('.', '/')
```
Comment (scpike) [inline test/mimemagic_test.rb]: Had to change to `sub` because there are dots in "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"

**Comment by minad** [comment]
thx! are you fine if we drop 1.8.7 support?

**Comment by scpike** [comment]
@minad definitely fine with not supporting 1.8.7. I don't use that version personally, and the paperclip project doesn't support it.

**Comment by minad** [comment]
ok, dropped it :)

**Pull request merged**